owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt

```diff
@@ -1059,50 +1060,40 @@ class FileDisplayActivity : FileActivity(),
         if (!uiResult.data.isNullOrEmpty()) {
             var pos = 0
-            val fragmentManager = supportFragmentManager
+            var posDialog = data.lastIndex
             data.asReversed().forEachIndexed { index, file ->
                 val countDownValue = index + 1
 
-                val customDialog = FileAlreadyExistsDialog.Builder()
-                    .setTitle(
-                        this.getString(
-                            if (file.isFolder) {
-                                R.string.folder_already_exists
-                            } else {
-                                R.string.file_already_exists
-                            }
-                        )
-                    )
-                    .setDescription(
-                        this.getString(
-                            if (file.isFolder) {
-                                R.string.folder_already_exists_description
-                            } else {
-                                R.string.folder_already_exists_description
-                            }, file.fileName
-                        )
-                    )
-                    .setCheckboxText(this.getString(R.string.apply_to_all_conflicts, countDownValue.toString()))
-                    .build()
-
-
-                if (countDownValue > 1) {
-                    customDialog.checkboxVisible = true
-                }
+                val customDialog = FileAlreadyExistsDialog.newInstance(
+                    this.getString(
+                        if (file.isFolder) {
+                            R.string.folder_already_exists
+                        } else {
+                            R.string.file_already_exists
+                        }
+                    ),
+                    this.getString(
+                        if (file.isFolder) {
+                            R.string.folder_already_exists_description
+                        } else {
+                            R.string.file_already_exists_description
+                        }, file.fileName
+                    ),
+                    this.getString(R.string.apply_to_all_conflicts, countDownValue.toString()),
+                    countDownValue > 1
+                )
+                customDialog.isCancelable = false
+                customDialog.show(this.supportFragmentManager, CUSTOM_DIALOG_TAG)
 
-                fragmentManager?.let {
-                    customDialog.show(it, CUSTOM_DIALOG_TAG)
-                }
+                fileOperationsViewModel.openDialogs.add(customDialog)
 
-                openDialogs.add(customDialog)
 
                 customDialog.setDialogButtonClickListener(object : FileAlreadyExistsDialog.DialogButtonClickListener {
 
                     override fun onKeepBothButtonClick() {
-                        if (customDialog.isCheckBoxChecked) {
+                        if (fileOperationsViewModel.openDialogs[posDialog].isCheckBoxChecked) {
```

Comment:
You have the almost the me code in 3 places. ¿Can yo create a method to replace this code?

Reply:
Done